View Controllers/ListViewController.m

```diff
@@ -156,12 +214,49 @@ - (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForR
  - (NSInteger)tableView:(nonnull UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
      
      if (self.segmentIndex == 0) {
-         [self setView];
          return self.favorites.count;
      }
      else {
-         [self setView];
          return self.wishlist.count;
      }
- }
+     
+}
+
+- (void)didAddFavorite:(NSNotification*)notification {
+    GMSPlace* place = (GMSPlace*)notification.object;
+    
+    // add to the favorites
+    NSMutableArray<GMSPlace*>* favorites = [NSMutableArray arrayWithArray:self.favorites];
+    [favorites insertObject:place atIndex:0];
+    self.favorites = (NSArray*)favorites;
+    
+    // reload table
+    [self.tableView reloadData];
+    
+    // hide default label
+    if(self.segmentIndex == 0)
+    {
+        self.tableView.hidden = NO;
+        self.defaultView.hidden = YES;
+    }
+}
+
+- (void)didAddToWishlist:(NSNotification*)notification {
+    GMSPlace* place = (GMSPlace*)notification.object;
+    
+    // add to the favorites
+    NSMutableArray<GMSPlace*>* wishlist = [NSMutableArray arrayWithArray:self.wishlist];
```

Comment:
i loooove that we're strongly typing our arrays! awesome :) 


Here, let's do the same thing we did with  [[NSArray arrayWithObject:place] arrayByAddingObjectsFromArray:self.favorites]